lms/static/js/Markdown.Editor.js

```diff
@@ -1040,107 +1057,144 @@
         // isCancel is false if we are going to keep the text.
         var close = function (isCancel) {
             util.removeEvent(doc.body, "keydown", checkEscape);
-            var text = input.value;
+            var url = urlInput.value.trim();
+            var description = descInput.value.trim();
+
+            // reset the form to show no errors.
+            urlInput.classList.remove('has-error');
+            urlErrorMsg.style.display = 'none';
+            descInput.classList.remove('has-error');
+            descErrorMsg.style.display = 'none';
 
             if (isCancel) {
-                text = null;
+                url = null;
             }
             else {
                 // Fixes common pasting errors.
-                text = text.replace(/^http:\/\/(https?|ftp):\/\//, '$1://');
+                url = url.replace(/^http:\/\/(https?|ftp):\/\//, '$1://');
                 // doesn't change url if started with '/' (local)
-                if (!/^(?:https?|ftp):\/\//.test(text) && text.charAt(0) != '/') {
-                    text = 'http://' + text;
+                if (!/^(?:https?|ftp):\/\//.test(url) && url.charAt(0) !== '/') {
+                    url = 'http://' + url;
                 }
             }
+            var isValidUrl = /^((?:http|https|ftp):\/{2}|\/)[^]+$/.test(url),
+                isValidDesc = (
+                    descInput.checkValidity() &&
+                    (descInput.required ? description.length : true)
+                );
+
+            if ((isValidUrl && isValidDesc) || isCancel) {
+                dialog.parentNode.removeChild(dialog);
+                callback(url, description);
+            } else {
+                var error_count = 0;
+                if (!isValidUrl) {
+                    urlInput.classList.add('has-error');
+                    urlErrorMsg.style.display = 'inline-block';
+                    error_count += 1;
+                } if (!isValidDesc) {
+                    descInput.classList.add('has-error');
+                    descErrorMsg.style.display = 'inline-block';
+                    error_count += 1;
+                }
 
-            dialog.parentNode.removeChild(dialog);
+                document.getElementById('wmd-editor-dialog-form-errors').textContent = [
+                    error_count,
+                    error_count === 1 ? gettext('error found in form.') : gettext('errors found in form.'),

```

Comment:
It is hard for translators to handle partial sentences like these. Can we make them into full sentences? If not, please add a translator comment so that they know how these strings are used.